api/scpca_portal/models/project.py

```diff
@@ -329,7 +329,13 @@ def create_computed_files(
 
         def on_get_project_file(future):
             if computed_file := future.result():
-                computed_file.process_computed_file(clean_up_output_data, update_s3)
+                computed_file.save()
+
+                if update_s3:
+                    computed_file.upload_s3_file()
+                if clean_up_output_data:
+                    computed_file.clean_up_local_computed_file()
```

Comment:
Because `ComputedFile::upload_s3_file` initiates the s3 uploading by spawning off a subprocess (using the subprocess module), it should be possible from a process scheduling perspective for the `ComputedFile::clean_up_local_computed_file` to be called directly afterwards and to corrupt the s3 upload. How should we handle this?

Reply:
In theory yes, but the `subprocess` module in the python standard library is blocking so its not a problem.